tests/mockserver.py

```diff
@@ -222,6 +224,14 @@ def ssl_context_factory(keyfile='keys/localhost.key', certfile='keys/localhost.c
          )
 
 
+def broken_ssl_context_factory(keyfile='keys/localhost.key', certfile='keys/localhost.crt', cipher_string='DEFAULT'):
+    factory = ssl_context_factory(keyfile, certfile)
+    ctx = factory.getContext()
+    ctx.set_options(SSL.OP_CIPHER_SERVER_PREFERENCE | SSL.OP_NO_TLSv1_2)
```

Comment:
TLS 1.2 has it's own always enabled ciphers so I disable it.

Twisted recommends using `CertificateOptions` instead of `DefaultOpenSSLContextFactory`, and its API is cleaner, but it requires file-like objects for certs and also I don't know if all supported Twisted versions support this API.

Reply:
`CertificateOptions` was there at least in twisted 9.0.0 (https://twistedmatrix.com/documents/9.0.0/api/twisted.internet.ssl.CertificateOptions.html), so it should be fine to use. But I have no idea what I'm talking about, or what is better in this particular case :)